sklearn/tests/test_discriminant_analysis.py

```diff
@@ -148,6 +148,18 @@ def test_lda_transform():
     assert_raise_message(NotImplementedError, msg, clf.transform, X)
 
 
+def test_lda_n_components():
+    clf = LinearDiscriminantAnalysis(solver="svd", n_components=2)
+    assert_warns_message(UserWarning, "'n_components' is not less than the "
+                         "number of classes, setting '_max_components' to "
```

Comment:
just say "setting number of components to" rather than reference a private attribute